src/librustc_metadata/decoder.rs

```diff
@@ -645,6 +645,16 @@ impl<'a, 'tcx> CrateMetadata {
         self.get_impl_data(id).trait_ref.map(|tr| tr.decode((self, tcx)))
     }
 
+    /// Iterates over all the stability attributes in the given crate.
+    pub fn get_lib_features(&self) -> Vec<(ast::Name, Option<ast::Name>)> {
+        let mut features: Vec<_> = self.root
+            .lib_features
+            .decode(self)
+            .collect();
+        features.sort_unstable_by_key(|f| f.0.as_str());
```

Comment:
It would be better to do the sorting before encoding; then you don't have to sort every time when decoding.